src/iterators.jl

```diff
@@ -83,9 +83,83 @@ function reinit!(cc::CellCache, i::Int)
+ - `fc.grid :: AbstractGrid`: grid information used in iteration
+ - `fc.topology :: ExclusiveTopology`: topology information used in iteration
+
+**Methods with `FaceCache`**
+ - `reinit!(cache::FaceCache, this_face::FaceIndex, neighbor_face::FaceIndex)`: reinitialize [`FaceCache`](@ref)
+
+See also [`FaceIterator`](@ref).
+"""
+struct FaceCache{X,G<:AbstractGrid}
+    this_coords::Vector{X}
+    neighbor_coords::Vector{X}
+    this_face::ScalarWrapper{Int}
+    neighbor_face::ScalarWrapper{Int}
+    orientation_info::InterfaceOrientationInfo
+    # grid and topology information needed for iteration
+    grid::G
+    topology::ExclusiveTopology
+end
+
+function FaceCache(grid::Grid{dim,C,T}, topology::ExclusiveTopology) where {dim,C,T}
+    N = nnodes_per_cell(grid)
+    this_coords = zeros(Vec{dim,T}, N)
+    neighbor_coords = zeros(Vec{dim,T}, N)
+    return FaceCache(this_coords, neighbor_coords, ScalarWrapper(-1), ScalarWrapper(-1), InterfaceOrientationInfo(false, 0), grid, topology)
+end
+
+function FaceCache(dh::DofHandler{dim}, topology::ExclusiveTopology) where {dim}
+    N = nnodes_per_cell(get_grid(dh))
+    this_coords = zeros(Vec{dim, get_coordinate_eltype(get_grid(dh))}, N)
+    neighbor_coords = zeros(Vec{dim, get_coordinate_eltype(get_grid(dh))}, N)
+    return FaceCache(this_coords, neighbor_coords, ScalarWrapper(-1), ScalarWrapper(-1), InterfaceOrientationInfo(false, 0), get_grid(dh), topology)
+end
+
+function reinit!(cache::FaceCache, this_face::FaceIndex, neighbor_face::Union{Nothing, FaceIndex} = nothing)
+    resize!(cache.this_coords, nnodes_per_cell(cache.grid, this_face[1]))
+    get_cell_coordinates!(cache.this_coords, cache.grid, this_face[1])
+    cache.this_face[] = this_face[2]
+    if !isnothing(neighbor_face)
+        resize!(cache.neighbor_coords, nnodes_per_cell(cache.grid, neighbor_face[1]))
+        get_cell_coordinates!(cache.neighbor_coords, cache.grid, neighbor_face[1])
+        cache.neighbor_face[] = neighbor_face[2]
+        return cache
+    end
+    cache.neighbor_face[] = -1 
+    return cache
+end
+
 # reinit! FEValues with CellCache
 reinit!(cv::CellValues, cc::CellCache) = reinit!(cv, cc.coords)
 reinit!(fv::FaceValues, cc::CellCache, f::Int) = reinit!(fv, cc.coords, f)
+reinit!(fv::FaceValues, fc::FaceCache, here::Union{Nothing, Bool} = nothing) = begin
+    isnothing(here) && (fc.neighbor_face != -1) && @warn "Current facet has a neighbor, consider using `here::Bool` to specify which facet to reninit"    
+    (isnothing(here) || here) && reinit!(fv, fc.this_coords, fc.this_face)
+    (!isnothing(here) && here) || reinit!(fv, fc.neighbor_coords, fc.neighbor_face)
+end
+reinit!(iv::InterfaceValues, fc::FaceCache) = begin
+    @assert fc.neighbor_face == -1 "Current face does not belong to an interface or is on boundary"    
+    reinit!(iv, fc.this_coords, fc.this_face, fc.neighbor_coords, fc.neighbor_face)
+end
```

Comment:
So if I revert to `InterfaceCache`/`InterfaceIterator` should I revert this also or let `FaceValues` be initialized by `InterfaceCache`?
@KnutAM